contrib/node/src/python/pants/contrib/node/tasks/javascript_style.py

```diff
@@ -75,32 +84,75 @@ def _is_javascriptstyle_dir_valid(self, javascriptstyle_dir):
     return True
 
   @memoized_method
-  def _install_javascriptstyle(self, javascriptstyle_dir):
-    with pushd(javascriptstyle_dir):
+  def _bootstrap_default_eslinter(self, bootstrap_dir):
+    with pushd(bootstrap_dir):
       result, yarn_add_command = self.execute_yarnpkg(
-        args=['install'],
+        args=['add', 'eslint'],
         workunit_name=self.INSTALL_JAVASCRIPTSTYLE_TARGET_NAME,
         workunit_labels=[WorkUnitLabel.PREP])
       if result != 0:
-        raise TaskError('Failed to install javascriptstyle\n'
+        raise TaskError('Failed to install eslint\n'
                         '\t{} failed with exit code {}'.format(yarn_add_command, result))
-    javascriptstyle_bin_path = os.path.join(javascriptstyle_dir, 'bin', 'cli.js')
-    return javascriptstyle_bin_path
+    return bootstrap_dir
 
-  def _run_javascriptstyle(self, target, javascriptstyle_bin_path, files, fix=False):
-    args = [javascriptstyle_bin_path]
+  @memoized_method
+  def _install_eslint(self, bootstrap_dir):
+    """Install the ESLint distribution.
+
+    :rtype: string
+    """
+    with pushd(bootstrap_dir):
+      result, yarn_install_command = self.execute_yarnpkg(
+        args=['install'],
+        workunit_name=self.INSTALL_JAVASCRIPTSTYLE_TARGET_NAME,
+        workunit_labels=[WorkUnitLabel.PREP])
+      if result != 0:
+        raise TaskError('Failed to install ESLint\n'
+                        '\t{} failed with exit code {}'.format(yarn_install_command, result))
+
+    self.context.log.debug('Successfully installed ESLint to {}'.format(bootstrap_dir))
+    return bootstrap_dir
+
+  def _get_target_ignore_patterns(self, target):
+    ignore_path = next((source for source in target.sources_relative_to_buildroot()
```

Comment:
since we are only always look for a single ignore path, maybe it is more readable to construct the path and check for it in the sources?  The usage of next() seems to be unnecessary.